eppo_client/client.py

```diff
@@ -36,261 +34,187 @@ def __init__(
             callback=config_requestor.fetch_and_store_configurations,
         )
         self.__poller.start()
+        self.__evaluator = Evaluator(sharder=MD5Sharder())
 
     def get_string_assignment(
-        self, subject_key: str, flag_key: str, subject_attributes=dict()
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[SubjectAttributes] = None,
+        default=None,
     ) -> Optional[str]:
-        try:
-            assigned_variation = self.get_assignment_variation(
-                subject_key, flag_key, subject_attributes, VariationType.STRING
-            )
-            return (
-                assigned_variation.typed_value
-                if assigned_variation is not None
-                else assigned_variation
-            )
-        except ValueError as e:
-            # allow ValueError to bubble up as it is a validation error
-            raise e
-        except Exception as e:
-            if self.__is_graceful_mode:
-                logger.error("[Eppo SDK] Error getting assignment: " + str(e))
-                return None
-            raise e
+        return self.get_assignment_variation(
+            subject_key,
+            flag_key,
+            subject_attributes,
+            VariationType.STRING,
+            default=default,
+        )
+
+    def get_integer_assignment(
+        self,
+        subject_key: str,
+        flag_key: str,
+        subject_attributes: Optional[SubjectAttributes] = None,
+        default=None,
+    ) -> Optional[float]:
```

Comment:
The function sounds like it returns an integer, but the type hinting indicates that it's a float?

Also if integer assignments get their own function, should float/double assignments get a separate function? Are we planning to deprecate the more ambiguous numeric assignment method?

Reply:
Good catch, yes this should be `int`

I argued we should call it `get_float_assignment` for the numeric stuff but @leoromanovsky and @giorgiomartini0 prefered `numeric` so sticking with that